frontend/lib/src/util/UriUtil.test.ts

```diff
@@ -251,27 +251,52 @@ describe("getPossibleBaseUris", () => {
     {
       description: "pathnames with a single part",
       pathname: "foo",
-      expectedBasePaths: ["foo", ""],
+      expectedBasePaths: ["", "foo"],
     },
     {
       description: "pathnames with two parts",
       pathname: "foo/bar",
-      expectedBasePaths: ["foo/bar", "foo"],
+      expectedBasePaths: ["", "foo/bar", "foo"],
     },
     {
       description: "pathnames with more than two parts",
       pathname: "foo/bar/baz/qux",
-      expectedBasePaths: ["foo/bar/baz/qux", "foo/bar/baz"],
+      expectedBasePaths: [
+        "",
+        "foo/bar/baz/qux",
+        "foo/bar/baz",
+        "foo/bar",
+        "foo",
+      ],
+    },
+    {
+      description: "pathnames with preference defined",
+      baseHref: "/foo/bar/",
+      pathname: "foo/bar/baz/qux",
+      expectedBasePaths: [
+        "foo/bar",
+        "foo/bar/baz/qux",
+        "foo/bar/baz",
+        "foo",
+        "",
```

Comment:
If the `baseHref` is `/foo/bar/`, why is `foo` appearing standalone?

Reply:
So the first option would be foo/bar (which SHOULD work because it's been specified).

If it does _not_ work, it will iterate through all the possible paths, progressively removing a directory.

Reply:
> If it does not work, it will iterate through all the possible paths, progressively removing a directory.

Ahh ok, so this is the old behavior which acts as a fallback here, right? If so, it might make sense to add your explanation as a comment to the code for future readers 🙂